source/androidx.media2/media2-session/Transforms/Metadata.xml

```diff
@@ -75,4 +75,24 @@
     <!-- Same method as base class, except with a covariant return type -->
     <remove-node path="/api/package[@name='androidx.media2.session']/class[@name='MediaController.Builder']/method[@name='setConnectionHints' and count(parameter)=1 and parameter[1][@type='android.os.Bundle']]" />
 
+    <attr
+        path="/api/package[@name='androidx.media2.session']/interface[@name='IMediaController']"

```

Comment:
What error(s) do these fix?  It looks like they rename existing types, which will break any existing users:

```
#### Removed Type AndroidX.Media2.Session.IMediaController

#### New Type: AndroidX.Media2.Session.IMediaControllerInterface

```csharp
public interface IMediaControllerInterface : Android.OS.IInterface, Android.Runtime.IJavaObject, Java.Interop.IJavaPeerable, System.IDisposable {
	// properties
	public static string Descriptor { get; }
	// methods
	public virtual void OnAllowedCommandsChanged (int p0, AndroidX.VersionedParcelable.ParcelImpl p1);
	public virtual void OnBufferingStateChanged (int p0, AndroidX.VersionedParcelable.ParcelImpl p1, int p2, long p3, long p4, long p5);
	public virtual void OnChildrenChanged (int p0, string p1, int p2, AndroidX.VersionedParcelable.ParcelImpl p3);
	public virtual void OnConnected (int p0, AndroidX.VersionedParcelable.ParcelImpl p1);
...
}
```

Reply:
Errors:

```
./generated/androidx.media2.media2-session/obj/Release/monoandroid12.0/generated/src/AndroidX.Media2.Session.IMediaController.cs(2118,24): error CS0260: Missing partial modifier on declaration of type 'MediaController'; another partial declaration of this type exists [./generated/androidx.media2.media2-session/androidx.media2.media2-session.csproj::TargetFramework=MonoAndroid12.0]
./generated/androidx.media2.media2-session/obj/Release/monoandroid12.0/generated/src/AndroidX.Media2.Session.IMediaSession.cs(3204,24): error CS0260: Missing partial modifier on declaration of type 'MediaSession'; another partial declaration of this type exists [./generated/androidx.media2.media2-session/androidx.media2.media2-session.csproj::TargetFramework=MonoAndroid12.0]
./generated/androidx.media2.media2-session/obj/Release/monoandroid12.0/generated/src/AndroidX.Media2.Session.IMediaSessionService.cs(396,24): error CS0260: Missing partial modifier on declaration of type 'MediaSessionService'; another partial declaration of this type exists [./generated/androidx.media2.media2-session/androidx.media2.media2-session.csproj::TargetFramework=MonoAndroid12.0]
./generated/androidx.media2.media2-session/obj/Release/monoandroid12.0/generated/src/AndroidX.Media2.Session.MediaController.cs(20,3): error CS0579: Duplicate 'global::Android.Runtime.Register' attribute [./generated/androidx.media2.media2-session/androidx.media2.media2-session.csproj::TargetFramework=MonoAndroid12.0]
./generated/androidx.media2.media2-session/obj/Release/monoandroid12.0/generated/src/AndroidX.Media2.Session.MediaSession.cs(20,3): error CS0579: Duplicate 'global::Android.Runtime.Register' attribute [./generated/androidx.media2.media2-session/androidx.media2.media2-session.csproj::TargetFramework=MonoAndroid12.0]
./generated/androidx.media2.media2-session/obj/Release/monoandroid12.0/generated/src/AndroidX.Media2.Session.MediaController.cs(1448,34): error CS0102: The type 'MediaController' already contains a definition for '_members' [./generated/androidx.media2.media2-session/androidx.media2.media2-session.csproj::TargetFramework=MonoAndroid12.0]
./generated/androidx.media2.media2-session/obj/Release/monoandroid12.0/generated/src/AndroidX.Media2.Session.IMediaSessionService.cs(396,24): error CS0263: Partial declarations of 'MediaSessionService' must not specify different base classes [./generated/androidx.media2.media2-session/androidx.media2.media2-session.csproj::TargetFramework=MonoAndroid12.0]
./generated/androidx.media2.media2-session/obj/Release/monoandroid12.0/generated/src/AndroidX.Media2.Session.MediaSessionService.cs(20,3): error CS0579: Duplicate 'global::Android.Runtime.Register' attribute [./generated/androidx.media2.media2-session/androidx.media2.media2-session.csproj::TargetFramework=MonoAndroid12.0]
./generated/androidx.media2.media2-session/obj/Release/monoandroid12.0/generated/src/AndroidX.Media2.Session.MediaSession.cs(1150,34): error CS0102: The type 'MediaSession' already contains a definition for '_members' [./generated/androidx.media2.media2-session/androidx.media2.media2-session.csproj::TargetFramework=MonoAndroid12.0]
./generated/androidx.media2.media2-session/obj/Release/monoandroid12.0/generated/src/AndroidX.Media2.Session.MediaSessionService.cs(169,17): error CS0111: Type 'MediaSessionService' already defines a member called 'MediaSessionService' with the same parameter types [./generated/androidx.media2.media2-session/androidx.media2.media2-session.csproj::TargetFramework=MonoAndroid12.0]
./generated/androidx.media2.media2-session/obj/Release/monoandroid12.0/generated/src/AndroidX.Media2.Session.MediaSessionService.cs(139,34): error CS0102: The type 'MediaSessionService' already contains a definition for '_members' [./generated/androidx.media2.media2-session/androidx.media2.media2-session.csproj::TargetFramework=MonoAndroid12.0]
```

caused by:

```
path="/api/package[@name='androidx.media2.session']/class[@name='MediaController']"
path="/api/package[@name='androidx.media2.session']/interface[@name='IMediaController']"


	[Register ("androidx/media2/session/IMediaController", DoNotGenerateAcw=true)]
	public abstract class MediaController : Java.Lang.Object {

```

this is just one case.

Reply:
I think we want this instead, to suppress creating the new conflicting types:

```xml
<!-- Static fields on interfaces cause us to generate a 'MediaController' class to hold the static fields, but that conflicts with actual 'MediaController' class -->
<attr path="/api/package[@name='androidx.media2.session']/interface[@name='IMediaController']" name="no-alternatives">true</attr>
<attr path="/api/package[@name='androidx.media2.session']/interface[@name='IMediaSession']" name="no-alternatives">true</attr>
<attr path="/api/package[@name='androidx.media2.session']/interface[@name='IMediaSessionService']" name="no-alternatives">true</attr>
```

Reply:
Added supression. I need to find some time to read more about this.

Fixed